taichi/llvm/llvm_program.cpp

```diff
@@ -604,5 +604,18 @@ uint64_t *LlvmProgramImpl::get_ndarray_alloc_info_ptr(DeviceAllocation &alloc) {
     return (uint64_t *)cpu_device()->get_alloc_info(alloc).ptr;
   }
 }
+
+void LlvmProgramImpl::fill_ndarray(DeviceAllocation &alloc,
```

Comment:
@qiao-bo  nit: Do we need mutable reference `DeviceAllocation&`?

If not, let's make them const. Otherwise let's follow Google's style and use pointer

Reply:
@k-ye , no, we don't need mutable, I will change this in #3921. Thanks